src/adapter/src/coord/read_policy.rs

```diff
@@ -145,8 +145,18 @@ impl<S: Append + 'static> crate::coord::Coordinator<S> {
                 let mut read_capability: ReadCapability<_> = policy.into();
 
                 if let Some(timeline) = self.get_timeline(*id) {
+                    let initial_frontier = self
+                        .controller
+                        .compute(*compute_instance)
+                        .unwrap()
+                        .collection(*id)
+                        .unwrap()
+                        .read_capabilities
+                        .frontier()
+                        .to_owned();
                     let TimelineState { read_holds, .. } =
                         self.ensure_timeline_state(timeline).await;
+                    assert!(initial_frontier.less_equal(&read_holds.time));
```

Comment:
We probably want a more detailed error message like in the previous PR.

Reply:
Done! Now we'll be able to see in gory detail why the invariant does not hold!